pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -361,10 +401,29 @@ func loadIPSecKeys(r io.Reader) (uint8, error) {
 		ipSecKey.Spi = spi
 
 		if len(s) == 6+offset {
+			if ipSecKeysGlobal[s[5+offset]] != nil {
+				oldSpi = ipSecKeysGlobal[s[5+offset]].Spi
+			}
 			ipSecKeysGlobal[s[5+offset]] = ipSecKey
 		} else {
+			if ipSecKeysGlobal[""] != nil {
+				oldSpi = ipSecKeysGlobal[""].Spi
+			}
 			ipSecKeysGlobal[""] = ipSecKey
 		}
+
+		scopedLog.WithError(err).Warning("newtimer: oldSPI %u new spi %u", oldSpi, ipSecKey.Spi)
+		// Detect a version change and call cleanup routine to remove old
+		// keys after a timeout period. We also want to ensure on restart
+		// we remove any stale keys for example when a restart changes keys.
+		// In the restart case oldSpi will be '0' and cause the delete logic
+		// to run.
+		if oldSpi != ipSecKey.Spi {
+			go func() {
+				time.Sleep(linux_defaults.IPsecKeyDeleteDelay)
+				ipsecDeleteXfrmSpi(ipSecKey.Spi)
```

Comment:
Might be good to log after the sleep, but before deleting here? I think I'd want to know if somehow the forwarding is changed at an arbitrary point in future if I need to debug something going wrong here.

Reply:
Can add add a dbg statement.

Reply:
This shouldn't happen terribly often should it? Just wondering whether Info would be fine for this. Probably stick with Debug unless others pipe up supporting Info.

Reply:
Should only happen when keys change so Info might be nice.